ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp

```diff
@@ -0,0 +1,141 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <iterator>
+#include <limits>
+#include <type_traits>
+
+#include "ngraph/check.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            namespace details
+            {
+                template <bool check>
+                using Required = typename std::enable_if<check, bool>::type;
+
+                template <typename It>
+                struct IsRandomAccessIt
+                {
+                    static constexpr bool value =
+                        std::is_same<typename It::iterator_category,
+                                     std::random_access_iterator_tag>::value;
+                };
+
+                template <typename... Args>
+                using void_t = void;
+
+                template <typename, typename = size_t>
+                struct is_complete : std::false_type
+                {
+                };
+
+                template <typename T>
+                struct is_complete<T, decltype(sizeof(T))> : std::true_type
+                {
+                };
+
+            } // namespace details
+
+            /// @brief Span should mimic std::span
+            template <typename Element>
+            class Span
```

Comment:
This is not a request to change but I only should mention that this class can operate only with vectors and arrays.
We cannot use any random access iterator because the container must be in a contiguous space in memory (like std::vector or C array).

Reply:
That is the idea to work on continues part of memory but the idea, to create some `Multispan`, is worth to consider.